### ISSUE TITLE
Guest subscribers issue - STOREID is null

### ISSUE THREAD

**Issue by epson121** [mailchimp/mc-magento2#75]
**Preconditions**

Magento 2.1.5
Mailchimp 1.0.22
Multiple website environment

**Steps to reproduce**

    Subscribe to a newsletter as a guest

**Actual and Expected result**

**Expected:**
	Mailchimp list gets updated with a new subscriber

**Actual:**
	Mailchimp list does not get updated with a new subscriber.

_CURL request params (email address omitted):_
	`{"status":"subscribed","email_address":"xxxxxx@xxxx.xxx","merge_fields":{"CGROUP":"NOT LOGGED IN","STOREID":null},"status_if_new":"subscribed"}`

_Mailchimp response:_
	`{"type":"http://developer.mailchimp.com/documentation/mailchimp/guides/error-glossary/","title":"Invalid Resource","status":400,"detail":"The resource submitted could not be validated. For field-specific details, see the 'errors' array.","instance":"","errors":[{"field":"merge_fields.STOREID","message":"Data did not match any of the schemas described in anyOf."}]}
`
So basically, STOREID is null for guests, and is not being received on Mailchimps end. When I try to update the code, and force-set the ID to a correct value (for testing purposes), it is being transferred fine.

Can you help with this one?

**Comment by gonzaloebiz** [comment]
Hi @epson121 

see the associated commit, and define the STOREID merge tag in mailchimp

Best

**Comment by epson121** [comment]
Thank you for the response. Can you clarify how to define the STOREID merge tag in mailchimp, since we're still getting errors:

`MailChimpLogger.INFO: No such entity with email = xxxxxx@xxxx.xx, websiteId = 1`

**Comment by gonzaloebiz** [comment]
https://kb.mailchimp.com/lists/manage-contacts/set-default-merge-values-for-a-list
